set.mm

```diff
@@ -80882,354 +81079,532 @@ C Fn ( ( S i^i dom F ) u. { z } ) ) $=
+       class is a subset of ` dom F ` .  Avoids the axiom of replacement.
+       (Contributed by Scott Fenton, 21-Apr-2011.)  (Proof shortened by Scott
+       Fenton, 17-Nov-2024.) $)
+    wfrdmcl $p |- ( X e. dom F -> Pred ( R , A , X ) C_ dom F ) $=
+      ( c2nd ccom cwrecs cfrecs df-wrecs eqtri frrdmcl ) ABCDGHZECABDIABNJFABDK
+      LM $.
+    $( $j usage 'wfrdmcl' avoids 'ax-rep'; $)
+  $}
+
+  ${
+    wfrfung.1 $e |- F = wrecs ( R , A , G ) $.
+    $( The well-ordered function generator generates a function.  Avoids the
+       axiom of replacement.  (Contributed by Scott Fenton, 21-Apr-2011.)
+       (Revised by Mario Carneiro, 26-Jun-2015.)  (Revised by Scott Fenton,
+       17-Nov-2024.) $)
+    wfrfung $p |- ( ( R We A /\ R Se A ) -> Fun F ) $=
+      ( wwe wse wa wfr wpo wfun wefr adantr wor weso sopo syl simpr c2nd ccom
+      cwrecs cfrecs df-wrecs eqtri fprfung syl3anc ) ABFZABGZHABIZABJZUHCKUGUIU
+      HABLMUGUJUHUGABNUJABOABPQMUGUHRABCDSTZCABDUAABUKUBEABDUCUDUEUF $.
+    $( $j usage 'wfrfung' avoids 'ax-rep'; $)
+
+    $( Show without using the axiom of replacement that the restriction of the
+       well-ordered recursion generator to a predecessor class is a set.
+       (Contributed by Scott Fenton, 18-Nov-2024.) $)
+    wfrresex $p |- ( ( ( R We A /\ R Se A ) /\ X e. dom F ) ->
+    ( F |` Pred ( R , A , X ) ) e. _V ) $=
+      ( wwe wse wa wfr wpo w3a cdm wcel cpred cres cvv wefr adantr wor weso syl
+      sopo simpr 3jca c2nd ccom cwrecs cfrecs df-wrecs eqtri fprresex sylan ) A
+      BGZABHZIZABJZABKZUOLECMNCABEOPQNUPUQURUOUNUQUOABRSUNURUOUNABTURABUAABUCUB
+      SUNUOUDUEABCDUFUGZECABDUHABUSUIFABDUJUKULUM $.
+    $( $j usage 'wfrresex' avoids 'ax-rep'; $)
+
+    $( A weak version of ~ wfr2g which is useful for proofs that avoid the
+       Axiom of Replacement.  (Contributed by Scott Fenton, 30-Jul-2020.)
+       (Proof shortened by Scott Fenton, 18-Nov-2024.) $)
+    wfr2ag $p |- ( ( ( R We A /\ R Se A ) /\ X e. dom F ) ->
+    ( F ` X ) = ( G ` ( F |` Pred ( R , A , X ) ) ) ) $=
+      ( wwe wse wa cdm wcel cfv cpred cres c2nd ccom co wfr adantr simpr cwrecs
+      wpo w3a wceq wefr wor weso sopo syl cfrecs df-wrecs eqtri fpr2a sylan cvv
+      3jca wfrresex opco2 eqtrd ) ABGZABHZIZECJZKZIZECLZECABEMNZDOPZQZVGDLVBABR
+      ZABUBZVAUCVDVFVIUDVBVJVKVAUTVJVAABUESUTVKVAUTABUFVKABUGABUHUISUTVATUPABCV
+      HECABDUAABVHUJFABDUKULUMUNVEEVGDVCUOVBVDTABCDEFUQURUS $.
+    $( $j usage 'wfr2ag' avoids 'ax-rep'; $)
+  $}
+
+  ${
+    wfrfun.1 $e |- R We A $.
+    wfrfun.2 $e |- R Se A $.
+    wfrfun.3 $e |- F = wrecs ( R , A , G ) $.
+    $( The well-ordered function generator generates a function.  (Contributed
+       by Scott Fenton, 21-Apr-2011.)  (Revised by Mario Carneiro,
+       26-Jun-2015.)  (Proof shortened by Scott Fenton, 17-Nov-2024.) $)
+    wfrfun $p |- Fun F $=
+      ( wwe wse wfun wfrfung mp2an ) ABHABICJEFABCDGKL $.
+    $( $j usage 'wfrfun' avoids 'ax-rep'; $)
```

Comment:
Is this inference a worthwhile addition to the closed form wfrfung ?  I would say "no".  If a second form is added to the closed form wfrfung, it would rather be the deduction form.

Reply:
Fair enough - I was just doing it because the old wfrfun was used later on. I've revised that proof now.